frontend/src/components/explorer/gemBrowser/MapsAvailable.vue

```diff
@@ -65,32 +70,19 @@ export default {
       model: state => state.models.model,
       mapsAvailable: state => state.maps.availableMaps,
     }),
-    mapAvailableLimited() {
-      /* eslint-disable vue/no-side-effects-in-computed-properties */
-      // TODO: move this into vuex
-      if (Object.keys(this.mapsAvailable).length === 0) {
-        return null;
+  },
+  methods: {
+    routeSVGmap(svgId, dimension) {
+      console.log('Zup');
+      if (this.viewerSelectedID) {
+        this.$router.push({ name: 'viewer',
+          params: { model: this.model.short_name, map_id: svgId, reload: true },
+          query: { dim: dimension, search: this.viewerSelectedID, sel: this.viewerSelectedID } });
+      } else {
+        this.$router.push({ name: 'viewer',
+          params: { model: this.model.short_name, map_id: svgId },
+          query: { dim: dimension } });
       }
```

Comment:
It's great that we moved this logic from markup to JavaScript.

1. Is `reload: true` necessary? 
2. It would be nice if this is further refactored to reduce repetition, for example:

```javascript
const params = { model: .. }
const query = { model: .. }

if (this. this.viewerSelectedID) {
  params.reload = true;
  query.search = ..
  ..
}
this.$router.push({ name: 'viewer', params, query });
```

Reply:
It looks like the `reload` stuff is obsolete since [this commit](https://github.com/MetabolicAtlas/MetabolicAtlas/commit/e39ab79dd98dd61eb1fdb02349eb002e85e2eb8e#diff-eec96ec11596d102e2cf2efc9389cbe113c1bb28c5eb7db5249808a83abcab03). It also looks like it's not directly related to this issue. I can open a new issue for removing `reload` related stuff in the whole project, but I think it's safe for you to omit it from this file for now if you want to @inghylt .